core/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -427,13 +427,13 @@ public void onTimeout(TimeValue timeout) {
             try {
                 future.get();
                 logger.trace("successfully waited for cluster state with version {} (current: {})", clusterStateVersion,
-                    observer.observedState().getVersion());
+                    clusterService.state().getVersion());
```

Comment:
nit: it looks like we can just capture the version here and forget about the entire cluster state